internal/kafka/internal/handlers/kafka.go

```diff
@@ -50,8 +50,8 @@ func (h kafkaHandler) Create(w http.ResponseWriter, r *http.Request) {
 			ValidateKafkaClaims(ctx, ValidateUsername(), ValidateOrganisationId()),
 			ValidateCloudProvider(ctx, h.service, &kafkaRequestPayload, h.providerConfig, "creating kafka requests"),
 			ValidateKafkaPlan(ctx, h.service, h.kafkaConfig, &kafkaRequestPayload),
+			validateKafkaBillingModel(ctx, h.service, h.kafkaConfig, &kafkaRequestPayload),
```

Comment:
I moved the order of the validation to do it before the cloud account id and marketplace validation as it is less resource intensive than the later.

I think it shouldn't impact the behavior. If you think otherwise let me know.